core/src/main/java/org/n52/iceland/request/AbstractServiceRequest.java

```diff
@@ -175,6 +180,18 @@ public Locale getRequestedLocale() {
         return LocaleHelper.fromString(getRequestedLanguage());
     }
 
+    public String getOriginalRequest() {
```

Comment:
For me the only reason to have these `has...` and `isSet...` methods was the absence of an `Optional` in the JDK. You could have used Guava's version, but that pulls in a big dependency just for `Optional`. But now with Java 8, I don't see anything that would speak against using it just like any other `java.util` class.
We should discuss this tomorrow in the Huddle...